xarray/core/indexing.py

```diff
@@ -345,20 +428,31 @@ def __init__(self, array, key=None):
             Array indexer. If provided, it is assumed to already be in
             canonical expanded form.
         """
-        if key is None:
-            key = (slice(None),) * array.ndim
-        self.array = array
-        self.key = key
+        # We need to avoid doubly wrapping.
+        if isinstance(array, type(self)):
```

Comment:
Does this actually come up? I thought we were careful to avoid this (but maybe not)